sdk/core/azure_core_amqp/build.rs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corp. All Rights Reserved.
+
+// cspell: words rustc AZSDK
+use rustc_version::version;
+
+fn main() {
+    let version = match version() {
+        Ok(version) => version.to_string(),
+        Err(_) => "unknown".to_owned(),
+    };
+    println!("cargo:rustc-env=AZSDK_RUSTC_VERSION={version}");
```

Comment:
Is this needed? We specify an MSRV in `azure_core` already (will probably move to `typespec`). If anything, we should have a general build function every crate can/should use with just a single call. I'd open a bug if you think it important to keep this information, but with an MSRV I really don't think it is.

Reply:
No - it came in as a copy from eventhubs.

And it is needed in eventhubs to capture the compiler version for the user-agent string (eventhubs has to generate the user-agent privately).